README.md

```diff
@@ -20,76 +22,174 @@ This document defines an extension to the service types supported in the [DID Co
+    "https://ipfs.io/ipfs/QmYbmJiuUtuNjFA7L7HVCgCCK2GFbxZ2VttFgTvurrfZff"
   ]
 }
 ```
 
-## Security considerations
+A Typescript snippet showing how to derive the service ID for the example document above is shown below:
+
+```ts
+import { blake2AsU8a } from '@polkadot/util-crypto'
+import * as multibase from 'multibase'
+import canonicalize from 'canonicalize'
+
+const doc = `
+{
+  "polkadot:b0a8d493285c2df73290dfb7e61f870f/slip44:434": {
+    "EJDj2GKnx89HTzUkGW8Rk9RoYUmAJHPM8aacWFp3fi1gYUQ": {
+      "description": "Personal account"
+    }
+  },
+  "polkadot:411f057b9107718c9624d6aa4a3f23c1/slip44:2086": {
+    "4nvZhWv71x8reD9gq7BUGYQQVvTiThnLpTTanyru9XckaeWa": {
+      "description": "Council account",
+      "proof": {
+        "scheme": "schorr-ristretto-25519",
+        "digest": "blake2b-256",
+        "signature": "0xae5f4d97dd67d45f8c6cb7e4977b9bdd4ccdd14db341995ba5074bccbe27c004a17bcf4a53e1e6a1eaac135c5f2b492e7d84dbbe4d80c221d3caed915f7b1286"
+      }
+    },
+    "4tMSjvHfWBNQw4tYGvkbRp7BBpwAB6S24LuMDcASYgnGnRTM" : {
+      "description": "Personal account"
+    }
+  },
+  "polkadot:91b171bb158e2d3848fa23a9f1c25182/slip44:354": {
+    "15BQbTH5bKH63WCXTMPxbmpnWeXKpfuTKbpDkfFLXMPvpxD3": {
+      "description": "Personal account"
+    }
+  }
+}
+`
+
+async function main() {
+  const jsonInput = JSON.parse(doc)
+  const canonicalJson = canonicalize(jsonInput)
+  const buffer = Buffer.from(canonicalJson as any)
+  const hash = blake2AsU8a(buffer)
+  const encoded = multibase.encode('base64urlpad', hash)
+  console.log(Buffer.from(encoded).toString('utf-8'))
+}
+
+main()
+```
+
+### Proof Registry
+
+As there is no extensive collection for definitions of hashing and digital signature schemes that are used in the crypto interactions, here is a list, meant to be extended, of all hashing and digital signature schemes that can be found in the `proof` object of a linked account, as well as the definition of what is the payload to generate such proof.
+
+#### Digital Signature Schemes
+
+* `schorr-ristretto-25519`: the Schnorr signature on Ristretto-compressed Ed25519 points. They are typically identified as `sr25519` in the Polkadot ecosystem.
+* `eddsa-25519`: the EdDSA digital signature scheme based on Ed25519 points. They are typically identified as `ed25519` in the Polkadot ecosystem.
+* `ecdsa-secp256k1`: the EDDSA digital signature scheme based on the secp256k1 curve. They are typically identified as `ecdsa` in the Polkadot ecosystem and are the default scheme for Ethereum and Bitcoin accounts.
+
+#### Hashing Schemes
+
+* `blake2b-256`: Blake2b hash with 256 bit output. They are the most common choice for Polkadot networks.
+* `keccak-256`: Hashing scheme used by Ethereum with 256 bit output.
+
+### Signature Generation and Verification
+
+Wherever possible, accounts exposed as part of this spec SHOULD include the optional `proof` property that provides evidence of control of such an account by the DID subject.
```

Comment:
The only point I see to have a proof in the json is to ensure that a user has access to the account that is being provided. But I don't think that it is necessary to include the signature into the json, dapp can just check it by itself and notify the user on ui/ux level.

Reply:
Agree. Will remove that from the spec. It makes everyone's life so much easier.